chef/cookbooks/corosync/recipes/service.rb

```diff
@@ -111,23 +111,42 @@
           "#{block_corosync_file} and run chef-client."
   end
 
-  # this service will remove the blocking file on proper shutdown
-  template "/etc/init.d/#{corosync_shutdown}" do
-    source "corosync-shutdown.init.erb"
-    owner "root"
-    group "root"
-    mode 0755
-    variables(
-      :service_name => node[:corosync][:platform][:service_name],
-      :block_corosync_file => block_corosync_file
-    )
-  end
+  if node.platform_family != "suse" || node.platform_version.to_f < 12.0
+    # this service will remove the blocking file on proper shutdown
+    template "/etc/init.d/#{corosync_shutdown}" do
+      source "corosync-shutdown.init.erb"
+      owner "root"
+      group "root"
+      mode 0755
+      variables(
+        :service_name => node[:corosync][:platform][:service_name],
+        :block_corosync_file => block_corosync_file
+      )
+    end
 
-  # Make sure that any dependency change is taken into account
-  bash "insserv #{corosync_shutdown} service" do
-    code "insserv #{corosync_shutdown}"
-    action :nothing
-    subscribes :run, resources(:template=> "/etc/init.d/#{corosync_shutdown}"), :delayed
+    # Make sure that any dependency change is taken into account
+    bash "insserv #{corosync_shutdown} service" do
```

Comment:
It's not about enabling the service, but about making sure the dependency chain is taken into account if there's any dependency change. (Btw, that's not new code, just indentation change here)